client/components/Nav.jsx

```diff
@@ -138,14 +243,7 @@ class Nav extends React.PureComponent {
               </button>
               <li className="nav__dropdown-item">
                 <button
-                  onClick={() => {
-                    if (!this.props.unsavedChanges) {
-                      this.props.newProject();
-                    } else if (this.props.warnIfUnsavedChanges()) {
-                      this.props.newProject();
-                    }
-                    this.setDropdown('none');
-                  }}
+                  onClick={this.handleNew}
                   onFocus={this.handleFocus.bind(this, 'file')}
```

Comment:
are binding these functions inline also an issue? from your original issue i thought this was the issue, rather than the inline anonymous functions

Reply:
They are certainly both performance issues, but binding with an argument will convert less cleanly into one variable for each argument to the function:

```javascript
this.handleFocusFile = this.handleFocus.bind(this, 'file');
this.handleFocusEdit = this.handleFocus.bind(this, 'edit');
```

Still, this should ideally be done and I'd say both are encompassed by the original issue (Despite `bind` being mentioned)

Reply:
@manaswinidas got it. thanks!